### PR TITLE
Agustin/2500-support-to-deploy-a-subset-of-docker-compose-services-in-the-okteto-manifest

### LINKED ISSUE #2500: Support to deploy a subset of docker compose services in the okteto manifest
**Is your feature request related to a problem? Please describe.**
My okteto manifest looks like this:
```
deploy:
   compose: docker-compose.yml
...
``` 
but I need to deploy only a subset of services
 
**Describe the solution you'd like**
Something like this:
```
deploy:
   compose:
      file: docker-compose.yml
      services:
        - a
        - b
...

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
Fixes #2500

## Proposed changes

- Accept the schema proposed here Support to deploy a subset of docker compose services in the okteto manifest #2500

### CODE REVIEW

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -266,10 +266,24 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
...
 			deployOptions.Manifest.Name = deployOptions.Name
 		}
-		if deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.Compose != nil && deployOptions.Manifest.Deploy.Compose.Stack != nil {
-			deployOptions.Manifest.Deploy.Compose.Stack.Name = deployOptions.Name
+		if deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.ComposeSection != nil && deployOptions.Manifest.Deploy.ComposeSection.Stack != nil {
+			deployOptions.Manifest.Deploy.ComposeSection.Stack.Name = deployOptions.Name
 		}
 	}
+
+	if deployOptions.Manifest != nil && deployOptions.Manifest.Deploy != nil && deployOptions.Manifest.Deploy.ComposeSection != nil {
```
Comment (jLopezbarb) [inline cmd/deploy/deploy.go:274]: We could include this if clause in the one above. If a composeSection has any manifest defined it must have a Stack field not nil

Comment (jLopezbarb) [inline cmd/deploy/deploy.go:274]: We could include this if clause in the one above. If a composeSection has any manifest defined it must have a Stack field not nil

File: cmd/deploy/deploy.go [hunk 2]
```diff
@@ -266,10 +266,24 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
...
+		var manifestDeclaredServicesToDeploy []string
```
Comment (jLopezbarb) [inline cmd/deploy/deploy.go:275]: Can we create a function and add tests.

Comment (jLopezbarb) [inline cmd/deploy/deploy.go:275]: Can we create a function and add tests.

**Review by jLopezbarb — APPROVED** [review]

**Comment by pchico83** [comment]
@AgustinRamiroDiaz would it be hard to only deploy the `endpoints` and `volumes` referenced by the subset of services?

**Comment by AgustinRamiroDiaz** [comment]
> @AgustinRamiroDiaz would it be hard to only deploy the `endpoints` and `volumes` referenced by the subset of services?

I don't think so. If it is hard I think it means we need a refactor on how we are flowing the information, so anyway it's a good idea to try. I'll create an issue to work on it

**Pull request merged**